### ISSUE TITLE
OpenNox Hates the Wizard in War01a.

### ISSUE THREAD

**Issue by KILAH4716** [opennox/opennox#544]
https://youtu.be/J1jT3HiHcFo?t=139

I have greater details of what happens in the description, but tl;dr : If you load War01a in OpenNox, the Wizard in the beginning cutscene is not invisible for whatever odd reason. If you load from Nox Reloaded, save and attempt to load the save from OpenNox, the game session(?) crashes, and you're sent back to the menu, and you can't even play as a new Warrior again until you do a hard reset of the game.

Meanwhile, if you play the cutscene through Nox Reloaded, the Wizard has no issues, and you can load from the Gauntlet from either game with no problems. I have not experimented with what happens after.

**Comment by dennwc** [comment]
Likely something is wrong with Invisibility spell cast by script.

Do you still have the save that fails on OpenNox? It's worth testing it separately.

**Comment by KILAH4716** [comment]
If I don't have the saves, it's not difficult to produce a new one. Uno momento.

**Comment by KILAH4716** [comment]
[Nox War01a Problems.zip](https://github.com/noxworld-dev/opennox/files/9761579/Nox.War01a.Problems.zip)
TestWarNR is the save file generated by Nox Reloaded.
TestWarON is generated by OpenNox.

**Comment by blmarket** [comment]
As someone worked a bit on scripts recently, I'd like to take a look at this.

**Comment by dennwc** [comment]
Not sure if related, but I bisected new game crash to 29830c114dbb5774db3c08ac974ea5adf4c2b36c. After fixing it the game doesn't crash on the save file.

I, will force push the fixed history soon (because we need all past commits to work for future bisects).

Sorry for doing it before you attempted to fix it @blmarket. The problem was because of this:
```c
uint32_t nox_xxx_waypointsHead_2523752 = 0;
// changed to
nox_waypoint_t* nox_xxx_waypointsHead_2523752 = 0;
```

This causes lines like this to use different offset during pointer arithmetic:
```c
*(uint32_t*)(nox_xxx_waypointsHead_2523752 + 488) = v2;
```

The fix was to explicitly cast to old type:
```c
*(uint32_t*)((uint32_t)nox_xxx_waypointsHead_2523752 + 488) = v2;
```

So @blmarket, next time please change the type to `void*` first, make it compile, then switch to correct type (as explained [here](https://github.com/noxworld-dev/opennox/blob/dev/docs/refactoring.md#changing-integer-variables-to-pointers)). Sorry for missing it during the code review :sweat_smile:

**Comment by blmarket** [comment]
I see, my bad. should have put more attention to those type changes.

**Comment by KILAH4716** [comment]
I want to add to this issue:
I just attempted to reload a Nox Reloaded save in OpenNox AFTER the Wizard cutscene in Chapter 2 from War01a, and the game crashes back to the main menu, so there's more going on here, and I don't understand what the problem might be.

**Comment by Evengard** [comment]
We really need bisecting theese regressions...

**Comment by dennwc** [comment]
@Evengard one way to do so is to actually help us do it :stuck_out_tongue:

**Comment by KILAH4716** [comment]
I truly believe issue #566 is directly intertwined with this issue, and I got more proof.

In the last section of Chapter 6, I saved in the middle of the necromancer fight while he was invisible, and tried loading with the exact same issue: I crash to the main menu, I'm unable to play that save again, and I need to load from a different save file. Invisibility is definitely the core of the problem, which is going to make Chapter 7 for the Warrior and Conjurer living hell.
